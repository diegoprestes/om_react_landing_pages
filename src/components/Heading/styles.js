import styled, { css } from 'styled-components';

const mediaFont = (theme) => css`
  @media ${theme.media.sm} {
    font-size: ${theme.fonts.sizes['4xl']};
  }
`;

const titleSize = {
  sm: (theme) => css`
    font-size: ${theme.fonts.sizes.md};
  `,
  md: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
  `,
  lg: (theme) => css`
    font-size: ${theme.fonts.sizes.xl};
  `,
  xl: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
    ${mediaFont(theme)};
  `
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)}
  `}
`;
