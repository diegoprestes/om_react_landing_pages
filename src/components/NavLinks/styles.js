import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    align-items: center;

    @media ${theme.media.sm} {
      flex-flow: row wrap;
    }
  `}
`;
