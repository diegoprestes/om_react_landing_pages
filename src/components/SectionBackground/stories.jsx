import SectionBackground from '.';

export default {
  title: 'SectionBackground',
  component: SectionBackground,
  args: {
    children: (
      <div>
        <h1>SectionBackground</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium delectus tempora
          temporibus obcaecati ea. Magnam commodi totam, enim placeat molestias, mollitia harum
          dignissimos adipisci, quibusdam deleniti cupiditate architecto? Soluta, itaque?
        </p>
      </div>
    )
  },
  argTypes: {
    children: { type: '' }
  }
};

export const Template = (args) => {
  return (
    <div>
      <SectionBackground {...args} />
    </div>
  );
};
