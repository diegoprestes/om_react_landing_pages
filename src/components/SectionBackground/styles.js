import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, hasBackground }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${hasBackground && containerBackgroundActivate(theme)}
  `}
`;
