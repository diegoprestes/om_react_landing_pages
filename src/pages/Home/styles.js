import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme, background }) => css`
    background: ${background || theme.colors.primaryColor};
  `}
  color: white;
`;
