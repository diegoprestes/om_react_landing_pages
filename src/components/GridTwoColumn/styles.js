import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      grid-template-columns: 1fr 2fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      margin-bottom: 0;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
