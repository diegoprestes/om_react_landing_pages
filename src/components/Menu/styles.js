import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as SectionBackground } from '../SectionBackground/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 5;
    background: ${theme.colors.white};
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    transition: all 300ms ease-in-out;
    ${visible && menuVisible(theme)}

    > ${SectionContainer} {
      display: grid;
      grid-template-columns: 1fr;
      grid-template-rows: 1fr;
      height: 100vh;
      overflow-y: auto;
      align-items: center;
      padding: 0 ${theme.spacings.lg};
    }

    > ${SectionBackground} {
      min-height: 100vh;
      display: flex;
      align-items: center;
    }

    & ${Heading} {
      display: flex;
      justify-content: center;
      overflow-y: none;
      padding-bottom: ${theme.spacings.lg};
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.sm} {
      position: static;
      height: auto;
      border-bottom: 1px solid ${theme.colors.mediumGray};
      ${menuVisible(theme)};

      > ${SectionContainer} {
        display: block;
        height: auto;
        padding: 0;
      }

      & ${Heading} {
        display: block;
        padding: 0;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.xl} 0;

    @media ${theme.media.sm} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: block;
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    cursor: pointer;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }

    @media ${theme.media.sm} {
      display: none;
    }
  `}
`;
