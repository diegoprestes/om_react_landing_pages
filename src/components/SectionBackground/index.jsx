import PropTypes from 'prop-types';
import SectionContainer from '../SectionContainer';
import * as Styled from './styles';

const SectionBackground = ({ children, hasBackground = false }) => {
  return (
    <Styled.Container hasBackground={hasBackground}>
      <SectionContainer>{children}</SectionContainer>
    </Styled.Container>
  );
};

SectionBackground.propTypes = {
  children: PropTypes.node.isRequired,
  hasBackground: PropTypes.bool
};

export default SectionBackground;
