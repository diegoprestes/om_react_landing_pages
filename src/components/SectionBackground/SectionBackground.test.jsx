import SectionBackground from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

describe('<SectionBackground />', () => {
  it('should render with dark background', () => {
    const { container } = renderTheme(
      <SectionBackground hasBackground>
        <h1>children</h1>
      </SectionBackground>
    );
    expect(container).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with light background', () => {
    const { container } = renderTheme(
      <SectionBackground>
        <h1>children</h1>
      </SectionBackground>
    );
    expect(container).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
