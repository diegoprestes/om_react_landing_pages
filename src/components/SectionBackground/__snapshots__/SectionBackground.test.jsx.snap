// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with dark background 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #0A1128;
  color: #FFFFFF;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      children
    </h1>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with light background 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      children
    </h1>
  </div>
</div>
`;
